vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSPackageRegistry.java

```
@@ -0,0 +1,570 @@
+        // JcrPackageDefinition jDef = pack.getDefinition();
+        // if (jDef == null || !jDef.getId().isValid()) {
+        // continue;
+        // }
+        // packages.add(jDef.getId());
+        // }
+        // }
+        // }
+        return packageIds;
+    }
+
+    /**
+     * Returns the path of this package.this also includes the version, but
+     * never the extension (.zip).
+     *
+     * @param id
+     *            the package id
+     * @return the path of this package
+     */
+    public String getInstallationPath(PackageId id) {
+
+        String installPath;
+        installPath = getMappedPackage(id);
+        if (installPath == null) {
+            StringBuilder b = new StringBuilder("/");
+            if (id.getGroup().length() > 0) {
+                b.append(id.getGroup());
+                b.append("/");
+            }
+            b.append(id.getName());
+            String v = id.getVersion().toString();
+            if (v.length() > 0) {
+                b.append("-").append(v);
+            }
+            installPath = b.toString();
+        }
+
+        return installPath;
+    }
+
+    private String getMappedPackage(PackageId id) {
+        // TODO: @suess lookup if installpath is mapped
+        return null;
+    }
+
+    /**
+     * Creates a random package id for packages that lack one.
+     * 
+     * @return a random package id.
+     */
+    private static PackageId createRandomPid() {
+        return new PackageId("temporary", "pack_" + UUID.randomUUID().toString(), (String) null);
+    }
+
+    @Nonnull
+    @Override
+    public ExecutionPlanBuilder createExecutionPlan() {
+        return new ExecutionPlanBuilderImpl(this);
+    }
+
+    @Override
+    public void installPackage(Session session, RegisteredPackage pkg, ImportOptions opts, boolean extract) throws IOException, PackageException {
+
+        if (session == null){
+            log.error("Installation of packages only supported when session is set.");
```

maybe throw an exception here?